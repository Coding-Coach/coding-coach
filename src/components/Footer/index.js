import React, { Fragment } from 'react';
import { Link } from 'gatsby'

import CookiesBanner from '../Banner/Cookies';
import ScrollTop from '../Button/ScrollTop';

export default function Footer() {
  return (
    <Fragment>
      <footer className="container">
        <div className="pt-4 pb-8 border-t border-secondary-lightest clearfix">
          <ul className="float-right">
            <LinkItem to="/legal/terms-conditions">Terms & Conditions</LinkItem>
            <LinkItem to="/legal/cookies">Cookies</LinkItem>
            <LinkItem to="/legal/privacy">Privacy Policy</LinkItem>
            <LinkItem to="/legal/code-of-conduct">Code Of Conduct</LinkItem>
          </ul>
        </div>
      </footer>
      <CookiesBanner />
      <ScrollTop />
    </Fragment>
  );
}

function LinkItem({ children, to }) {
  return (
    <li className="mr-4 inline-block md:ml-4 md:mr-0">
      <Link
        to={to}
        className="text-body text-secondary-dark hover:underline hover:text-primary-light"
      >
        {children}
      </Link>
    </li>
  );
}
