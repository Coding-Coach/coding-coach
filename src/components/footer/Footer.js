import React from 'react';
import { Trans } from '@lingui/macro';

import Button from 'components/button/Button';

// @TODO: Bring it back when ready
// import DonateButton from './DonateButton';
import styles from './footer.module.scss';

function Footer({ onClickLegal, t }) {
  return (
    <footer className={styles.footerMain}>
      <div className={styles.footerInner}>
        <div className={styles.column}>{/* <DonateButton /> */}</div>

        <Button
          type="tertiary"
          typography="capitalize"
          size="small"
          onClick={() => onClickLegal('terms')}
          className={styles.footerLink}
        >
          <Trans id="footer.legal.terms" />
        </Button>
        <Button
          type="tertiary"
          typography="capitalize"
          size="small"
          onClick={() => onClickLegal('cookies')}
          className={styles.footerLink}
        >
          <Trans id="footer.legal.cookies" />
        </Button>
        <Button
          type="tertiary"
          typography="capitalize"
          size="small"
          onClick={() => onClickLegal('privacy')}
          className={styles.footerLink}
        >
          <Trans id="footer.legal.privacy" />
        </Button>
        <Button
          type="tertiary"
          typography="capitalize"
          size="small"
          onClick={() => onClickLegal('conduct')}
          className={styles.footerLink}
        >
          <Trans id="footer.legal.conduct" />
        </Button>
      </div>
    </footer>
  );
}

export default Footer;
