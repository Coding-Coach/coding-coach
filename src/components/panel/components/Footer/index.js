import PanelFooter from './PanelFooter';

export default PanelFooter;
