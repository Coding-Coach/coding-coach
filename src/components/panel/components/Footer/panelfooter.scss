
.footer {
  display: flex;

  &::first-child {
    margin-left: auto;
  }
}

.pullRight {
  margin-left: auto;
}