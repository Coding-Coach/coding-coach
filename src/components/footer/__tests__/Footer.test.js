import React from 'react';
import { render } from 'react-testing-library';
import Footer from '../Footer';

test('renders the footer element', () => {
  const { container } = render(<Footer />);

  expect(container.firstChild).toMatchSnapshot();
});
