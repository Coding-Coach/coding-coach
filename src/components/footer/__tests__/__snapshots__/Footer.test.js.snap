// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the footer element 1`] = `
<footer
  class="footerMain"
>
  <div
    class="footerInner"
  >
    <div
      class="column"
    />
    <button
      class="button tertiary small rounded capitalize"
    >
      footer.legal.terms
    </button>
    <button
      class="button tertiary small rounded capitalize"
    >
      footer.legal.cookies
    </button>
    <button
      class="button tertiary small rounded capitalize"
    >
      footer.legal.privacy
    </button>
    <button
      class="button tertiary small rounded capitalize"
    >
      footer.legal.conduct
    </button>
  </div>
</footer>
`;
